src/dsp_tools/commands/xmlupload/models/input_problems.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+list_separator = "\n    - "
+medium_separator = "\n----------------------------\n"
+
+
+@dataclass(frozen=True)
+class AllIIIFUriProblems:
+    """Information about all problems with IIIF URIs."""
+
+    problems: list[IIIFUriProblem]
+
+    def get_msg(self) -> str:
+        """Get a message describing all problems with the IIIF URIs."""
+        msg = "There were problems with the following IIIF URI(s):"
+        all_msg = [problem.get_msg() for problem in self.problems]
+        return msg + medium_separator + medium_separator.join(all_msg)
+
+
+@dataclass(frozen=True)
+class IIIFUriProblem:
+    """Information about a problem with an IIIF URI."""
+
+    uri: str
+    regex_has_passed: bool
+    status_code: int | None = None
+    raised_exception_name: str | None = None
+    original_text: str | None = None
```

Comment:
remark: it's ok to leave this as is now, but just because it caught my eye:  
the types of this class suggest that it would be more accurate to model this as two separate kinds of problems:

```python
type IIIFUriProblem = IIIFUriFormatProblem | IIIFServerNotReachableProblem

@dataclass(frozen=True)
class IIIFUriFormatProblem:
    uri: str
    # no boolean needed, this type indicates that the regex failed

@dataclass(frozen=True)
class IIIFServerNotReachableProblem:
    uri: str
    status_code: int
    exception_name: str
    original_text: str
    # no optionality needed, the type only gets instantiated when these params are non-optional
```

Not sure if there is a real advantage - but in terms of keeping separate concepts separate, this might make sense.  
Also, this would make it more natural to have `IIIFServerNotReachableProblem | None` be returned by `_make_network_call()`, which might not be a bad thing because you wouldn't have to pass around unspecified exceptions.

Reply:
Principally I like the suggestion to separate this cleanly, but there are some reasons why I kept i
You asked later if it makes sense to do a network call if the regex did not pass. I did this because we cannot be 100% sure that our regex does not fail valid URIs. If this were the case, then I would like to get notified, so that we can take a look and perhaps change the regex. I would also like to pass this info to the user, so that they don't get confused, why something valid is failing. Because this information is relevant to the message, keeping both information, in the same class is easier.